tests/unit/testplan/testing/multitest/test_result.py

```diff
@@ -58,14 +60,55 @@ def test_assertion_orders():
         "Report passed so far.",
     ]
     # pylint: disable=invalid-sequence-index
-    assertions = (
+    assertions = [
         entry
         for entry in mtest.report.flatten()
         if isinstance(entry, dict) and entry["meta_type"] == "assertion"
+    ]
+
+    for idx, desc in enumerate(expected):
+        assert desc == assertions[idx]["description"]
```

Comment:
how are these 2 lines different from original?

Reply:
the testcase raises an exception, so in the original for loop, the `actual` is enumerated but it is empty, so no assert statement is really executed. now the bug is fixed and we just enumerate the `expected`.